test/IntegrationTests/SqlServerCollection.cs

```diff
@@ -72,11 +74,36 @@ private async Task<TestcontainersContainer> LaunchSqlServerContainerAsync()
             .WithPortBinding(Port, DatabasePort)
             .WithEnvironment("SA_PASSWORD", Password)
             .WithEnvironment("ACCEPT_EULA", "Y")
-            .WithWaitStrategy(Wait.ForUnixContainer().UntilPortIsAvailable(DatabasePort));
+            .WithWaitStrategy(Wait.ForUnixContainer().UntilPortIsAvailable(DatabasePort))
+            .WithWaitStrategy(Wait.ForUnixContainer().UntilOperationIsSucceeded(DatabaseLoginOperation, 15));
 
         var container = databaseContainersBuilder.Build();
         await container.StartAsync();
 
         return container;
     }
+
+    private bool DatabaseLoginOperation()
+    {
+        try
+        {
+            using (var connection = new SqlConnection(GetConnectionString()))
```

Comment:
we can remove a little noise and inline `GetConnectionString()` 

```suggestion
            string connectionString = $"Server=127.0.0.1,{Port};User=sa;Password={Password};TrustServerCertificate=True;";
            using (var connection = new SqlConnection(connectionString))
```